iModelCore/ECDb/Tests/NonPublished/ClassViewsTests.cpp

```diff
@@ -0,0 +1,631 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See COPYRIGHT.md in the repository root for full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+#include "ECDbPublishedTests.h"
+
+USING_NAMESPACE_BENTLEY_EC
+BEGIN_ECDBUNITTESTS_NAMESPACE
+
+struct ClassViewsFixture : public ECDbTestFixture {};
+
+struct TestIssueListener: ECN::IIssueListener {
+    mutable std::vector<Utf8String> m_issues;
+    void _OnIssueReported(ECN::IssueSeverity severity, ECN::IssueCategory category, ECN::IssueType type, Utf8CP message) const override {
+        m_issues.push_back(message);
+    }
+    Utf8String const& GetLastError() const { return m_issues.back();}
+    Utf8String PopLastError() {
+        Utf8String str = m_issues.back();
+        m_issues.pop_back();
+        return str;
+    }
+    void Dump (Utf8CP listenerName) {
+        Utf8String cppCode;
+        for(auto it = m_issues.rbegin(); it != m_issues.rend(); ++it) {
+            cppCode.append(Utf8PrintfString("ASSERT_STREQ(\"%s\", %s.PopLastError().c_str());\r\n", (*it).c_str(), listenerName));
+        }
+        printf("%s", cppCode.c_str());
+    }
+    void Reset() { m_issues.clear(); }
+};
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST_F(ClassViewsFixture, prepare_view_and_check_validate_sql_and_data) {
+    auto testSchema = SchemaItem(R"xml(<?xml version="1.0" encoding="utf-8" ?>
+    <ECSchema
+            schemaName="test_schema"
+            alias="ts"
+            version="1.0.0"
+            xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+        <ECSchemaReference name='ClassViews' version='01.00.00' alias='classView' />
+        <ECEntityClass typeName="SchemaClassesView" description="" displayLabel="" modifier="Abstract">
+            <ECCustomAttributes>
+                <View xmlns="ClassViews.01.00.00">
+                    <Query>
+                        SELECT
+                            cd.ECInstanceId,
+                            cd.ECClassId,
+                            sc.Name SchemaName,
+                            cd.Name ClassName
+                        FROM meta.ECSchemaDef sc
+                            JOIN meta.ECClassDef cd ON cd.Schema.Id=sc.ECInstanceId
+                        GROUP BY sc.Name, cd.Name
+                        LIMIT 10
```

Comment:
What is a LIMIT going to do to the view?   Will it limit to 10 rows?

Reply:
Yes. It's just a SELECT query with all its capabilities. You can have UNION and LIMIT. It could be even a CTE or recursive query.